assets/js/sections/contesting.js

```diff
@@ -208,20 +214,35 @@ $('#start_date').change(function () {
 
 // On Key up check and suggest callsigns
 $("#callsign").keyup(function () {
-	var call = $(this).val();
+	var call = $(this).val().toUpperCase();
 	if (call.length >= 3) {
 
-		$.ajax({
-			url: 'lookup/scp',
-			method: 'POST',
-			data: {
-				callsign: $(this).val().toUpperCase()
-			},
-			success: function (result) {
-				$('.callsign-suggestions').text(result);
-				highlight(call.toUpperCase());
-			}
-		});
+        if (scp_data.status < 0 || scp_data.request != call.substr(0, scp_data.request.length)) {
+          scp_data.status = 0;
+          scp_data.request = call;
+          scp_data.data = [];
+          $.ajax({
+              url: 'lookup/scp',
+              method: 'POST',
+              data: {
+                  callsign: call
+              },
+              success: function (result) {
+                  scp_data.status = 1;
+                  scp_data.data = result.split(" ");
+
+                  var call = $("#callsign").val().toUpperCase(); // Might have changed while running the query...
+                  $('.callsign-suggestions').text(scp_data.data.filter((el) => el.startsWith(call)).join(' '));
```

Comment:
this will thow an error if `scp_data.data` is ever missing or falsey.  We could easily handle this scenario by doing this:

`$('.callsign-suggestions').text(scp_data.data?.filter((el) => el.startsWith(call)).join(' ') || []);`

We'll check if data exists on `scp_data` with an existential operator `(?)` and if it doesn't, we pass an empty arrow to the filter function.

Reply:
Your code returns an empty array for an empty string, where a string is expected (the `|| []` isn't passed to the `filter`). I guess `... || ""` is sufficient here.